rust/operator-binary/src/catalog/iceberg.rs

```diff
@@ -0,0 +1,35 @@
+use super::{config::CatalogConfig, FromTrinoCatalogError, ToCatalogConfig};
+use async_trait::async_trait;
+use stackable_operator::client::Client;
+use stackable_trino_crd::catalog::iceberg::IcebergConnector;
+
+pub const CONNECTOR_NAME: &str = "iceberg";
+
+#[async_trait]
+impl ToCatalogConfig for IcebergConnector {
+    async fn to_catalog_config(
+        &self,
+        catalog_name: &str,
+        catalog_namespace: Option<String>,
+        client: &Client,
+    ) -> Result<CatalogConfig, FromTrinoCatalogError> {
+        // Get config that is generated by attributed shared between Hive and Iceberg
+        let mut config = self
+            .common
+            .to_catalog_config(catalog_name, catalog_namespace, client)
+            .await?;
+
+        // As the `HiveAndIcebergCommonAttributes` struct can't know for which connector it is called
+        // we have to overwrite the connector name here
+        config.add_property("connector.name", CONNECTOR_NAME);
```

Comment:
Is `catalog_name` != `CONNECTOR_NAME`? This would be something belonging in the respective separated config for each catalog?

Reply:
The connector name is either `hive` or `iceberg`.
The catalog name is whatever the user calls the TrinoCatalog object (e.g. hive, foo or prod).
They can be the same but don't need to. I've use dozens of differently named catalogs in the past